lib/internal/worker.js

```diff
@@ -281,6 +288,13 @@ class Worker extends EventEmitter {
         null,
       hasStdin: !!options.stdin,
     }, transferList);
+    const loaderModule = require('internal/modules/esm/loader');
+    const hasCustomizations = loaderModule.hasCustomizations();
+    if (!isInternal && hasCustomizations) {
+      // - send the second side of the channel to the hooks thread
+      loaderModule.getHooksProxy().makeAsyncRequest(
+        '#registerWorkerClient', [toWorkerThread], toWorkerThread);
```

Comment:
```suggestion
      loaderModule.getHooksProxy().makeAsyncRequest(
        '#registerWorkerClient',
        new Array(toWorkerThread),
        toWorkerThread,
      );
```

`new Array` here enables V8 to create a more optimised array under the hood, sized exactly right instead of oversized for potential additions.

Also, nit for formatting.

Reply:
> `new Array` here enables V8 to create a more optimised array under the hood, sized exactly right instead of oversized for potential additions.

`new Array(toWorkerThread)` is not equivalent to `[toWorkerThread]`

Reply:
How so not? (Aside from the optimisation diff)

Reply:
```console
> JSON.stringify([2]) === JSON.stringify(new Array(2))
false
> JSON.stringify(new Array(2))
'[null,null]'
```

Reply:
In your example yes, but is `toWorkerThread` an integer? If not, what I said is accurate:

```console
> JSON.stringify(['foo']) === JSON.stringify(new Array('foo'))
true
> JSON.stringify(new Array('foo'))
'["foo"]'
```

Reply:
> the `Array` constructor is better […] than the array literal

I don't believe that's true, the explanation makes no sense (if V8 can optimize the `Array` constructor call, it can also optimize the array literal). In general, it's always better to chose syntax over runtime calls.

> In your example yes, but is `toWorkerThread` an integer? If not, what I said is accurate

I think that demonstrates why using the `Array` constructor call is a bad idea: it makes the code ambiguous to human readers, who probably won't know on the first read what type is `toWorkerThread`.

Reply:
@JakobJingleheimer I don't think that's how it works? (and checking this code in https://v8.github.io/tools/head/system-analyzer/index.html doesn't indicate anything changed recently)

Reply:
Let’s not argue over trivial things please? There’s nothing wrong with the original code.

Reply:
@GeoffreyBooth it's... interesting :D?

Reply:
It could matter, but not something to block, which is why I approved the PR 🙃

I'll try to find proof or lack thereof and open a follow-up if appropriate.